reco_utils/recommender/geoimc/GeoIMCdata.py

```diff
@@ -0,0 +1,162 @@
+import warnings
+from scipy.io import loadmat
+import pandas as pd
+import numpy as np
+from scipy.sparse import coo_matrix, isspmatrix_csr
+from sklearn.model_selection import train_test_split
+from sklearn import datasets
+from sklearn.preprocessing import normalize
+from numba import jit, prange
+
+from IPython import embed
+from .GeoIMCutils import length_normalize, reduce_dims
+
+
+class DataPtr():
+    """
+    Holds data and its respective indices
+    """
+
+    def __init__(self, data, entities):
+        assert isspmatrix_csr(data)
+
+        self.data = data
+        self.entities = entities
+        self.data_indices = None
+        self.entity_indices = [None, None]
+
+
+    def get_data(self):
+        if self.data_indices is None:
+            return self.data
+        return self.data[self.data_indices]
+
+
+    def get_entity(self, of="row"):
+        idx = 0 if of=="row" else 1
+        if self.entity_indices[idx] is None:
+            return self.entities[idx]
+        return self.entities[idx][self.entity_indices[idx]]
+
+
+class Dataset():
+    """
+    Base class that holds necessary (minimal) information needed
+    """
+
+    def __init__(
+            self,
+            name,
+            features_dim=0,
+            normalize=False,
+            target_transform=''
+    ):
+        """Initialize parameters
+
+        Args:
+            name (str): Name of the dataset
+            features_dim (uint): Dimension of the features. If not 0, PCA is performed
+                on the features as the dimensionality reduction technique
+            normalize (bool): Normalize the features
+            target_transform (str): Transform the target values. Current options are
+                'normalize' (Normalize the values), '' (Do nothing), 'binarize' (convert
+                the values using a threshold defined per dataset)
+
+        """
+        self.name = None
+        self.training_data = None
+        self.test_data = None
+        self.entities = None
+
+        self.features_dim = features_dim
+        self.feat_normalize = normalize
+        self.target_transform = target_transform
+
+
+    def load_data(self, path):
+        raise NotImplementedError(f"{self.name} should implement it")
+
+
+    def binarize_rating(self, v):
+        return 1 if v >=3 else 0
+
+
+    def normalize(self):
+        if self.feat_normalize:
+            for i in range(len(self.entities)):
+                if isspmatrix_csr(self.entities[i]):
+                    print(f"Normalizing CSR matrix")
```

Comment:
it would be nice to use logging, like we are doing in other parts of the code: https://github.com/microsoft/recommenders/blob/master/reco_utils/recommender/sar/sar_singlenode.py

Reply:
Resolved at b19a52b51b7df9dc272000623da4d82bfb7ceeff